akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/PredefinedFromStringUnmarshallers.scala

```diff
@@ -45,14 +45,17 @@ trait PredefinedFromStringUnmarshallers {
       }
     }
 
-  implicit val uuidFromStringUnmarshaller: Unmarshaller[String, UUID] =
-    Unmarshaller.strict { string =>
-      try UUID.fromString(string)
-      catch {
-        case e: IllegalArgumentException =>
-          throw new IllegalArgumentException(s"'$string' is not a valid UUID value", e)
-      }
+  implicit val uuidFromStringUnmarshaller: Unmarshaller[String, UUID] = {
+    val nilUuidString = "00000000-0000-0000-0000-000000000000"
+    val validUuidRegex = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}"
+
+    Unmarshaller.strict[String, UUID] { string =>
+      if (string == nilUuidString || string.matches(validUuidRegex))
```

Comment:
AFAICS this will compile the regex for each unmarshalling action, which I is expensive. Can we precompile the regex and hold on to the compiled pattern?

Reply:
You mean something like this?:
```
val validUuidPattern =
  "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}|00000000-0000-0000-0000-000000000000".r.pattern

  Unmarshaller.strict[String, UUID] { string =>
    if (validUuidPattern.matcher(string).matches)
      UUID.fromString(string)
    else
      throw new IllegalArgumentException(s"'$string' is not a valid UUID value")
  }
```

Reply:
Yes that looks like it'd avoid compiling the pattern for every unmarshalling!